### ISSUE TITLE
When pasting code into a function it doesn't indent well.

### ISSUE THREAD

**Issue by Storms-Engineering** [notepad-plus-plus/notepad-plus-plus#5923]
### Description of the Issue
When I past code into a blank indented function it doesn't automatically indent

### Steps to Reproduce the Issue

I have this empty function:
```
public function area_report()
{
       throw new Exception("Area report currently doesn't work");
}
```
and I paste this code into it below the ```throw new exception``` line:
```
if (isset($_GET['area']) && $_GET["area"] == "") {
	$query .= $GLOBALS['srbc_camps'] . ".area LIKE '%' ";
}
else {
	$values = array($_GET['area']);
	$query .= $GLOBALS['srbc_camps'] . ".area='%s' ";
}
```

It would be very nice if this indented properly.

### Expected Behavior
```
public function area_report()
{
	throw new Exception("Area report currently doesn't work");
	if (isset($_GET['area']) && $_GET["area"] == "") {
	$query .= $GLOBALS['srbc_camps'] . ".area LIKE '%' ";
	}
	else {
	$values = array($_GET['area']);
		$query .= $GLOBALS['srbc_camps'] . ".area='%s' ";
	}
}
````

### Actual Behavior
```
public function area_report()
{
	throw new Exception("Area report currently doesn't work");
	if (isset($_GET['area']) && $_GET["area"] == "") {
	$query .= $GLOBALS['srbc_camps'] . ".area LIKE '%' ";
}
else {
	$values = array($_GET['area']);
	$query .= $GLOBALS['srbc_camps'] . ".area='%s' ";
}
}
```

### Debug Information

Notepad++ v7.7.1   (64-bit)
Build time : Jun 16 2019 - 21:24:47
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 10 (64-bit)
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll

**Comment by Ekopalypse** [comment]
So what is your suggestion? Obviously npp is not an IDE and therefore can't now each programming language in detail. How should pasting something work in general? An additional way to paste based on the current column? Something else? To be honest, I think this is something a language depended plugin should solve.

**Comment by sasumner** [comment]
>I think this is something a language depended plugin should solve.

Or a script that, upon paste, queries the current column and then adjusts what is going to get pasted accordingly. :)

**Comment by Storms-Engineering** [comment]
Notepad++ already indents correctly if I past only one line of code, it just seems to be when I past multi lines.  Its true there could be a script, and I guess notepad++ isn't a full blown IDE.  Though for PHP programming I kind of use it as such... hahah.

**Comment by sasumner** [comment]
>Notepad++ already indents correctly if I past only one line of code

Notepad++ doesn't indent anything when you paste one line of code, or many lines of code.  All it does is paste what is there to paste, without changing anything otherwise.  Or do you have some hard evidence (provable example) to show that it "indents"?

**Comment by Storms-Engineering** [comment]
I don't have any evidence that it indents when pasting.  I only have evidence that it does indeed indent code.
If I type ```if (true)``` and then press enter type ```{``` and then press enter.  It will indent so my code looks like this:
```
if (true)
{
    | <-- cursor is here
```

**Comment by sasumner** [comment]
Well, what we're talking about here is something being changed via paste.  What you just mentioned I presume is coming from having this setting ticked:

![image](https://user-images.githubusercontent.com/30118311/61635350-5e92e080-ac61-11e9-972f-9e05fa574262.png)

**Comment by Storms-Engineering** [comment]
Yes that is true.  So this probably will not be added anytime, its true Notepad++ isn't a full blown IDE.

**Comment by dbareis** [comment]
I believe this is the same as my question "[Maintain Indent While Pasting Multiple Lines - Notepad++ Community](https://community.notepad-plus-plus.org/topic/25295/maintain-indent-while-pasting-multiple-lines)", anything code specific may be outside scope but at least to me the current npp behavior is wrong and simple to fix (assuming paste can be intercepted).
